### PR TITLE
Sequence page

### PR DESCRIPTION
**Pull request by mantecajc**
Dans cette branche, on a développé les pages séquence, séance, activité et atelier. 

Il n'est pas exactement le même que celui envoyé à PAD-EIG. Certains bugs ont été déjà corrigés ici.

### CODE REVIEW

File: theme/padplus/classes/output/core_renderer.php [hunk 1]
```diff
@@ -359,6 +386,75 @@ public function context_header_settings_menu() {
     }
     /*** PADPLUS END */
 
+    /**
+     * This is an optional menu that can be added to a layout by a theme. It contains the
+     * menu for the most specific thing from the settings block. E.g. Module administration.
+     *
+     * @return string
+     */
+    public function region_main_settings_menu() {
```
Comment (denier-epnak) [inline theme/padplus/classes/output/core_renderer.php:400]: We already have an overridden version in coursecat_settings_menu_padplus below. Need discussion

File: theme/padplus/javascript/add-folder-icon.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Allows us to add the folder icon to the page title when we are in a section/activity page.
+var sectionPageClass = document.querySelectorAll('.format-topics');
+
+if (sectionPageClass.length > 0) { // We check if the class is empty or not. It allows us to exclude workshop page.
```
Comment (denier-epnak) [inline theme/padplus/javascript/add-folder-icon.js]: is this essential?

File: theme/padplus/lang/en/theme_padplus.php [hunk 3]
```diff
@@ -83,11 +83,22 @@
 // Enrolment strings.
 $string['unenrolme'] = 'Unenrol me';
 
-// Categories & course page strings.
+// Categories, course, section & activity page strings.
 $string['addnewworkshop'] = 'Add new workshop';
-$string['btn-coursebox'] = 'View this course';
+$string['activity-session'] = '{$a->total} activity in this session:';
+$string['activities-session'] = '{$a->total} activities in this session:';
+$string['btn-coursebox'] = 'View this cours';
```
Comment (denier-epnak) [inline theme/padplus/lang/en/theme_padplus.php]: typo 'course'

File: theme/padplus/layout/columns2.php [hunk 4]
```diff
@@ -121,15 +121,31 @@
...
-// Group: current course and its sections. Options left out: 'badgesview', 'competencies'.
+// Group: current course. Options left out: 'badgesview', 'competencies'.
 fill_nav_from_menu_keys($nav, $padnav, ['coursehome', 'participants', 'grades']);
-fill_nav_from_menu_type($nav, $padnav, navigation_node::TYPE_SECTION);
-// Enable divider on coursehome since it comes after the main menu now.
+
+// Enable divider on coursehome since it comes after the main menu now & set a common text.
 $coursehome = $padnav->get('coursehome');
 if (is_object($coursehome)) {
+    if (course_is_workshop($this->page->course)) {
```
Comment (denier-epnak) [inline theme/padplus/layout/columns2.php]: Discussion : we have two different tests for workshop

File: theme/padplus/layout/columns2.php [hunk 5]
```diff
@@ -121,15 +121,31 @@
...
+        $nav->get('coursehome')->text = get_string('sidebar-sumamry-workshop', 'theme_padplus');
```
Comment (denier-epnak) [inline theme/padplus/layout/columns2.php]: In such cases, prefer ternary expressions (less indent = easier to read) :

$labelkey = course_is_workshop($this->page->course) ? 'sidebar-sumamry-workshop' : 'sidebar-summary-course';
$coursehome->text = get_string($labelkey, 'theme_padplus');

File: theme/padplus/lang/fr/theme_padplus.php [hunk 6]
```diff
@@ -83,11 +83,22 @@
...
+$string['next-activity'] = 'Activité suivante';
 $string['no-participants-enrolled'] = 'aucun stagiaire inscrit';
 $string['participants-enrolled'] = 'Stagiaires inscrits';
+$string['previous-session'] = 'Séance précédente';
+$string['previous-activity'] = 'Activité précédente';
+$string['progress-session'] = 'Progression :';
 $string['settings-category'] = 'Gestion de cette catégorie';
+$string['settings-activity'] = "Paramètres de l'activité";
+$string['sidebar-summary-course'] = 'Dans cette séquence';
+$string['sidebar-sumamry-workshop'] = 'Dans cet atelier';
```
Comment (denier-epnak) [inline theme/padplus/lang/fr/theme_padplus.php]: typo summary

File: theme/padplus/layout/columns2.php [hunk 7]
```diff
@@ -121,15 +121,31 @@
...
+    } else {
+        $nav->get('coursehome')->text = get_string('sidebar-summary-course', 'theme_padplus');
+    }
     $coursehome->set_showdivider(true, $coursehome->text);
 }
 
+// Unlink icons for grades & participants.
+$grades = $nav->get('grades');
```
Comment (denier-epnak) [inline theme/padplus/layout/columns2.php]: foreach(array('grades', 'participants') as $item) {
$navitem = $padnav->get($item);
if (is_object($navitem)) {
    $navitem->icon->pix = null;
}
}

File: theme/padplus/scss/post/layouts/_sidebar.scss [hunk 8]
```diff
@@ -44,6 +44,12 @@
             box-shadow: none !important;
         }
     }
+    [data-key = 'participants'],
+    [data-key = 'grades'] {
+        span {
+            margin-left: 29px;
```
Comment (denier-epnak) [inline theme/padplus/scss/post/layouts/_sidebar.scss]: isn't there some kind of indent stuff built-in Moodle flat navigation? I remember something like this

File: theme/padplus/scss/post/layouts/_sidebar.scss [hunk 9]
```diff
@@ -59,3 +65,12 @@
...
     margin-left: 6px;
 }
+
+// Icons
+.fa.fa-tachometer:before {
+    content: "\f009" !important; // icon fa-th-large fontawesome
+}
+
+.fa.fa-graduation-cap:before {
+    content: '\f07c' !important; // icon folder-open fontawesome
```
Comment (denier-epnak) [inline theme/padplus/scss/post/layouts/_sidebar.scss]: use theme icon system for overidding icons

File: theme/padplus/classes/output/format_topics_renderer.php [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+
+        /*** PADPLUS: Return section activities summary & session progress. */
+        $a = new stdClass;
+        $a->complete = $complete;
+        $a->total = $total;
+
+        // Output section activities summary.
+        $o = '';
+        $o .= html_writer::start_tag('div', array('class' => 'section-summary-activities pr-2 mdl-right'));
+        if ($total == 1) {
```
Comment (denier-epnak) [inline theme/padplus/classes/output/format_topics_renderer.php]: use ternary expression:
$label = ($total === 1) ? 'activity-session' : 'activities-session';

File: theme/padplus/classes/output/core/course_renderer.php [hunk 11]
```diff
@@ -420,4 +421,167 @@ public function course_info_box(stdClass $course) {
...
+        $completiondetails = \core_completion\cm_completion_details::get_instance($mod, $USER->id, $showcompletionconditions);
+        $ismanualcompletion = $completiondetails->has_completion() && !$completiondetails->is_automatic();
+
+        // Fetch activity dates.
+        $activitydates = [];
+        if ($course->showactivitydates) {
+            $activitydates = \core\activity_dates::get_dates_for_module($mod, $USER->id);
+        }
+
+        // If there is content AND a link, then display the content here
```
Comment (denier-epnak) [inline theme/padplus/classes/output/core/course_renderer.php:539]: Missing PADPLUS marker

File: theme/padplus/classes/output/core_renderer.php [hunk 12]
```diff
@@ -249,7 +249,20 @@ private function is_dashboard_page() {
...
+        $pagepath = $this->page->url->get_path();
+        $pageurl = $this->page->url->__toString();
+        return ($pagepath === '/course/view.php' && (strpos($pageurl, 'section') == true));
+    }
+
+    private function is_activity_page() {
+        $course = $this->page->course;
+        $pageurl = $this->page->url->__toString();
+        // We check if we are on the activity page of a course but we exclude workshop activity page.
+        return ((strpos($pageurl, '/mod/') == true) && (course_is_workshop($course) == false));
```
Comment (denier-epnak) [inline theme/padplus/classes/output/core_renderer.php]: it's a specific case: "mod/lesson" activity

File: theme/padplus/scss/post/components/_badges.scss [hunk 13]
```diff
@@ -16,7 +17,29 @@
...
 
     &.badge-pad-info {
+        align-items: center;
         background-color: map-get($badge-colors, "info");
+        display: flex;
+        width: fit-content;
+    }
+}
+
+.badge-success {
```
Comment (denier-epnak) [inline theme/padplus/scss/post/components/_badges.scss:31]: regression? &.badge-pad-workshop is not linked to badge-pad but to badge-success

File: theme/padplus/lang/en/theme_padplus.php [hunk 14]
```diff
@@ -83,11 +83,22 @@
...
+$string['next-session'] = 'Next session';
+$string['next-activity'] = 'Next activity';
 $string['no-participants-enrolled'] = 'no students enrolled';
 $string['participants-enrolled'] = 'Enrolled students';
+$string['previous-session'] = 'Previous session';
+$string['previous-activity'] = 'Previous activity';
+$string['progress-session'] = 'Progress:';
 $string['settings-category'] = 'Management of this category';
+$string['settings-activity'] = "Activity settings";
+$string['sidebar-summary-course'] = 'In this cours';
```
Comment (denier-epnak) [inline theme/padplus/lang/en/theme_padplus.php]: typo 'course'

File: theme/padplus/scss/post/pages/_categories.scss [hunk 15]
```diff
@@ -32,11 +32,11 @@
             border: 1px solid theme-color("pad-border");
             border-radius: 8px;
             display: flex;
-            height: 64px;
```
Comment (denier-epnak) [inline theme/padplus/scss/post/pages/_categories.scss]: Extract from PR

File: theme/padplus/classes/output/core_renderer.php (hunk 1 above)
Comment (denier-epnak) [reply theme/padplus/classes/output/core_renderer.php:400]: Simon

File: theme/padplus/javascript/add-folder-icon.js (hunk 2 above)
Comment (denier-epnak) [reply theme/padplus/javascript/add-folder-icon.js]: Javier

File: theme/padplus/lang/en/theme_padplus.php (hunk 3 above)
Comment (denier-epnak) [reply theme/padplus/lang/en/theme_padplus.php]: Javier

File: theme/padplus/layout/columns2.php (hunk 4 above)
Comment (denier-epnak) [reply theme/padplus/layout/columns2.php]: Simon

File: theme/padplus/layout/columns2.php (hunk 5 above)
Comment (denier-epnak) [reply theme/padplus/layout/columns2.php]: Simon

File: theme/padplus/classes/output/core/course_renderer.php (hunk 11 above)
Comment (denier-epnak) [reply theme/padplus/classes/output/core/course_renderer.php:539]: Javier

File: theme/padplus/scss/post/components/_badges.scss (hunk 13 above)
Comment (denier-epnak) [reply theme/padplus/scss/post/components/_badges.scss:31]: javier

File: theme/padplus/lang/en/theme_padplus.php (hunk 14 above)
Comment (denier-epnak) [reply theme/padplus/lang/en/theme_padplus.php]: javier

File: theme/padplus/scss/post/pages/_categories.scss (hunk 15 above)
Comment (denier-epnak) [reply theme/padplus/scss/post/pages/_categories.scss]: javier

File: theme/padplus/classes/output/core_renderer.php (hunk 12 above)
Comment (denier-epnak) [reply theme/padplus/classes/output/core_renderer.php]: Wait for discussion

File: theme/padplus/javascript/add-folder-icon.js (hunk 2 above)
Comment (mantecajc) [reply theme/padplus/javascript/add-folder-icon.js]: done

File: theme/padplus/scss/post/components/_badges.scss (hunk 13 above)
Comment (mantecajc) [reply theme/padplus/scss/post/components/_badges.scss:31]: J'en ai profité pour corriger le bug des margins (https://3.basecamp.com/5191032/buckets/24030658/todos/4739288153) et centrer verticalement le texte

File: theme/padplus/classes/output/format_topics_renderer.php (hunk 10 above)
Comment (denier-epnak) [reply theme/padplus/classes/output/format_topics_renderer.php]: javier

File: theme/padplus/scss/post/layouts/_sidebar.scss (hunk 9 above)
Comment (denier-epnak) [reply theme/padplus/scss/post/layouts/_sidebar.scss]: javier

File: theme/padplus/scss/post/layouts/_sidebar.scss (hunk 8 above)
Comment (denier-epnak) [reply theme/padplus/scss/post/layouts/_sidebar.scss]: simon

File: theme/padplus/layout/columns2.php (hunk 7 above)
Comment (denier-epnak) [reply theme/padplus/layout/columns2.php]: simon

File: theme/padplus/lang/fr/theme_padplus.php (hunk 6 above)
Comment (denier-epnak) [reply theme/padplus/lang/fr/theme_padplus.php:106]: javier

Comment (denier-epnak) [reply theme/padplus/lang/fr/theme_padplus.php]: simon

File: theme/padplus/scss/post/layouts/_sidebar.scss (hunk 8 above)
Comment (denier-epnak) [reply theme/padplus/scss/post/layouts/_sidebar.scss]: see indent/get_indent

**Pull request merged**